src/tools/miri/src/shims/foreign_items.rs

```diff
@@ -914,16 +914,6 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                 let x = this.read_scalar(x)?.to_f64()?;
                 let exp = this.read_scalar(exp)?.to_i32()?;
 
-                // Saturating cast to i16. Even those are outside the valid exponent range so
```

Comment:
I am trying to figure out why this was originally here so that I can justify why its being removed now.

I got to this commit: https://github.com/rust-lang/rust/commit/0743ed631efe63ae2de96847df7b7fce629127b6 
and now I'm stuck. Why were we clamping ldexp exponent to i16, and why don't we need to keep doing so now? I'm assuming its a weakness in the old APFloat API that has since been addressed in the new rustc_apfloat, but can you confirm that @wesleywiser ?

Reply:
FYI this is from Miri PR https://github.com/rust-lang/miri/pull/902

Reply:
> I'm assuming its a weakness in the old APFloat API that has since been addressed in the new rustc_apfloat, but can you confirm that @wesleywiser ?

Yes, exactly. The old APFloat API took ldexp as an `i16` but the new crate takes `i32` which matches what the code originally did before calling into APFloat. There's additional discussion why `i16` is sufficient for this parameter in the discussion here https://github.com/rust-lang/miri/pull/902#discussion_r312587825.

Reply:
I see. So in theory this could have stayed as-is, but since it was an artifact of a prior limitation, we're removing it. RIght?

Reply:
Yes, I could have just added an `.into()` to do the `i16 -> i32` conversion but since we're actually doing a `i32 -> i16 -> i32` conversion, it seemed cleaner to just remove the conversions entirely.

Reply:
Indeed, thanks for the nice cleanup. :)

Reply:
I can explain this one, it's https://github.com/llvm/llvm-project/commit/8606d013983d2098ba0d9ea9c2129a0b32a93e33 `[APFloat] Enlarge ExponentType to 32bit integer` from late 2019 - previously, miri was stuck with an `i32` and `rustc_apfloat` wanted `i16`, but they now both agree on `i32`.